component/conprof/store/store_test.go

```diff
@@ -239,6 +243,78 @@ func TestGenjiDBAddColumn(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestGenjiDBGCBug(t *testing.T) {
+	// related issue: https://github.com/genjidb/genji/issues/454
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "ngm-test-genjidb-gc.*")
+	require.NoError(t, err)
+	defer func() {
+		err := os.RemoveAll(tmpDir)
+		require.NoError(t, err)
+	}()
+	opts := badger.DefaultOptions(tmpDir).
+		WithMemTableSize(1024 * 1024).
+		WithNumLevelZeroTables(1).
+		WithNumLevelZeroTablesStall(2).WithValueThreshold(128 * 1024)
+
+	engine, err := badgerengine.NewEngine(opts)
+	require.NoError(t, err)
+	db, err := genji.New(context.Background(), engine)
+	require.NoError(t, err)
+	defer db.Close()
+	badgerDB := engine.DB
+
+	sql := "CREATE TABLE IF NOT EXISTS t (ts INTEGER PRIMARY KEY, data BLOB)"
+	err = db.Exec(sql)
+	require.NoError(t, err)
+	// triage the bug in https://github.com/genjidb/genji/issues/454.
+	err = db.Exec(sql)
+	require.NoError(t, err)
+
+	ts := time.Now().Unix()
+	for i := 0; i < 1000; i++ {
+		data := mockProfile()
+		sql = "INSERT INTO t (ts, data) VALUES (?, ?)"
+		err = db.Exec(sql, int(ts)+i, data)
+		require.NoError(t, err)
+	}
+
+	sql = "DELETE FROM t WHERE ts <= ?"
+	err = db.Exec(sql, int(ts)+1000)
+	require.NoError(t, err)
+
+	// Insert 1000 kv to make sure the deleted kv will been compacted.
+	for i := 1000; i < 2000; i++ {
+		data := mockProfile()
+		sql = "INSERT INTO t (ts, data) VALUES (?, ?)"
+		err = db.Exec(sql, int(ts)+i, data)
+		require.NoError(t, err)
+	}
+
```

Comment:
how about checking key count here?

Reply:
I use `Flatten` API to make sure the kv will be compacted then the deleted kv will be delete when compaction.

If I check the key count here, I can't make sure the compaction whether happend.